lib/wibox/widget/slider.lua

```diff
@@ -0,0 +1,508 @@
+        local handle_border_color = self._private.handle_border_color
+            or beautiful.slider_handle_border_color
+            or properties.handle_border_color
+
+        if handle_border_color then
+            cr:set_source(color(handle_border_color))
+        end
+
+        cr:set_line_width(handle_border_width)
+        cr:stroke()
+    end
+end
+
+function slider:fit(_, width, height)
+    -- Use all the space, this should be used with a constraint widget
+    return width, height
+end
+
+local event = {}
+
+local rect_order = {"x", "y", "width", "height"}
+
+-- Move the handle to the correct location
+local function move_handle(self, x, y, geo)
+    -- Fix the X position when used in a rotate or mirror widget
+
+    -- Remove the translation to bring {x, y} and the rect on the same origin
+    local rem_translate = geo.matrix_to_device*gmatrix.identity
+    rem_translate.x0, rem_translate.y0 = 0, 0
+
+    local rect = {rem_translate:transform_rectangle(
+        0,
+        0,
+        geo.width,
+        geo.height
+    )}
+
+    -- Floating point precision is here both too much and not enough
+    for k, v in ipairs(rect) do
+        rect[rect_order[k]] = util.round(v)
+    end
+
+    local px = rem_translate:transform_point(x, y)
+
+    -- If the transformation moved the rectangle away from {0,0}, compensate
+    if rect.x < 0 then
+        px = rect.width + px
+    end
+
+    self:set_value(get_value(self, rect.width, px))
+    self:emit_signal("widget::redraw_needed")
+end
+
+function event.mouse_press(self, x, y, button_id, _, geo)
+    if button_id ~= 1 then return end
+
+    move_handle(self, x, y, geo)
+
+    -- Get the "real" position of the widget
+    local wgeo = geo.drawable.drawable:geometry()
+    local wx = wgeo.x + geo.x
+    local wy = wgeo.y + geo.y
+
+    capi.mousegrabber.run(function(mouse)
+        if not mouse.buttons[1] then
+            self:emit_signal("widget::redraw_needed")
+            return false
+        end
+
+        -- Get the closest point to the widget
+        local abs_x, abs_y = mouse.x, mouse.y
+        abs_x = abs_x > (wx+geo.width) and wx+geo.width or (
+            abs_x < wx and wx
+        ) or abs_x
+        abs_y = abs_y > (wy+geo.height) and wy+geo.height or (
+            abs_y < wy and wy
+        ) or abs_y
+
+        -- Get the relative point
+        local rx, ry = abs_x - wx, abs_y - wy
+        move_handle(self, rx, ry, geo)
+
+        return true
+    end,"fleur")
+end
+
+--- Create a slider widget.
+-- @tparam[opt={}] table args
+-- @function wibox.widget.slider
+local function new(args)
+    local ret = base.make_widget(nil, nil, {
+        enable_properties = true,
+    })
+
+    util.table.crush(ret._private, args or {})
+
+    util.table.crush(ret, slider, true)
+
+    -- Change the cusor and redraw (in case there is hover and pressed colors)
+    ret:connect_signal("button::press", event.mouse_press)
+
+    return ret
+end
+
+function slider.mt:__call(...)
```

Comment:
This needs `_` as first argument (first argument being the metatable on which this function is called)